_hardware/cypherockx1.md

```diff
@@ -16,52 +16,138 @@ providerWebsite: https://cypherock.com/
+* On Powershell, setup Docker and compile:
 
-## Interface
+        PS C:\Users\LENEVO\Desktop\Cypherock\X1_wallet\Firmware\v0.4.773>  docker run -v ${pwd}/build:/out -it x1-wallet-app-env /bin/ash docker build . --file Dockerfile --tag x1-wallet-app-env
+        .
+        .
+        /home/build/Release # cmake -DDEV_SWITCH=OFF -DDEBUG_SWITCH=OFF -DSIGN_BINARY=OFF -DCMAKE_BUILD_TYPE:STRING=Release -DFIRMWARE_TYPE=Main -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=ON -DCMAKE_BUILD_PLATFORM:STRING=Device -G Ninja ../../
+        /home/build/Release # ninja
+        /home/build/Release # sha256sum Cypherock-Main.bin
+        54a2e548f3cb2f8a2f56671fe8b9df20ed90d084a9d68d435672e586e76c3903  Cypherock-Main.bin
 
-The device has a screen and one circular button that allows users to select the letters. It is quite tedious as the single button is very limited in the number of letters it can select with one push of this button. This is evident in the [video shown here](https://youtu.be/eD3CZMPumCk?t=204).
+* Compare from the Cypherock-Main.bin (unsigned) [release binary](https://github.com/Cypherock/x1_wallet_firmware/actions/runs/3521309283). Alternatively, compare using 256 bytes stripped off from Cypherock-Main-signed.bin
 
-## Reproducibility
+        MINGW64 ~/Desktop/Cypherock/X1_wallet/Firmware/v0.4.773 ((v0.4.773))
+        $ sha256sum.exe main-release-outputs/release/Cypherock-Main.bin
+        54a2e548f3cb2f8a2f56671fe8b9df20ed90d084a9d68d435672e586e76c3903 *main-release-outputs/release/Cypherock-Main.bin
+***Fingerprint (SHA256 hash) of released unsigned main X1 Wallet firmware binary and compiled main X1 Wallet firmware binary matches. Therefore, we conclude that the firmware is reproducible.***
 
-In its [2020 Bitcointalk.org ANN post](https://bitcointalk.org/index.php?topic=5225800.0), Cypherock claims that the hardware wallet **will** be open source. It also makes this claim on [its website](https://twitter.com/BitcoinWalletz/status/1462700172719243266). The product is still in pre-order. We were able to locate Cypherock's Github page, but there was [no corresponding repository](https://twitter.com/BitcoinWalletz/status/1462694293085118466) specifically for its hardware wallet. 
 
-But with that said, this hardware wallet has not yet been released.
\ No newline at end of file
+## Installing firmware updates
+> - The user has the flexibility to install a firmware update or not. The secure bootloader, which forms the root of trust, verifies if the firmware is a newer version than the currently installed version and [verifies the source of the firmware](https://github.com/Cypherock/x1_wallet_firmware/blob/main/docs/bootloader.md) (through 2 signatures).
```

Comment:
again a quote without source.

Reply:
Quotation mark not required. Source is already included with a hyperlink